Refactored for generic sphere radius

loraxipam opened the pull request:
I've refactored your work to create a generic haversine formula which allows you to use the haversine calculation on spheres other than the Earth. This _does_ break the Distance() function, though, so this really ought to be version 3.

Commit:
Refactored for generic sphere radius

Commit:
Add a missing math library import to the usage example

Commit:
Add a missing math library import to the usage example

Commit:
Spaces and tabs

Commit:
Remove extraneous planet_test.go

Commit:
apply gofmt

Commit:
Fix README

zacsketches commented:
@loraxipam ...love the PR.  I appreciate that you kept the playful nature of the the Examples we had already built, but fixed a few things like making the constants public and making the package more general.  It is a busy week for us here, but we will get a chance to look at the PR in detail before next Monday.  Really appreciate you taking some time to make this little tool just a bit better.

My intent is to merge the commit once we have a chance to look it over in detail.

Thank you, Zac.

Commit:
Merge branch 'master' of github.com:loraxipam/haversine

loraxipam commented:
Groovy, baby. Thanks for the quick response, Zac. No rush.

zacsketches committed:
task(go.mod): Initiated go.mod support

zacsketches committed:
Create LICENSE.txt

Updated the license file to the exact Github template so that it would pass the scan in pkg.go.dev.

zacsketches committed:
Update CHANGELOG.md

Reset version numbering to reflect minor version changes prior to a v.1.0.0 release. This update better communicates the package's maturing in anticipation of hosting it on pkg.go.dev.

zacsketches committed:
Create CONTRIBUTING.md

Added community contribution guidelines so we can start to grow some support for our community-based development of spatial measurement tools.

zacsketches commented:
@loraxipam ...I got into this PR today and realized we had not set this repo up for good community collaboration yet.  So I created a `dev` branch, added a [CONTIBUTING.md](https://github.com/FATHOM5/haversine/blob/dev/CONTRIBUTING.md), added `go.mod` so users can import the tools specific version, and a few other things.  

I am going to close this PR, because it is targeted at integration with master.  Please, pull the newest version of our repo and target your PR at the `dev` branch.

Thank you for getting me to take a look at this tool and update it for go modules, and creating a better open source tool.

Commit:
v0.3.0

Commit:
Straggler

Commit:
Merge branch 'dev'

Commit:
Merge branch 'master' of github.com:loraxipam/haversine

Commit:
Stragglers

Commit:
Merge branch 'master' of github.com:loraxipam/haversine

Commit:
Fixed go.mod module path

Commit:
Fix straggler import errors

Commit:
Merge branch 'master' of github.com:FATHOM5/haversine
